fe/fe-core/src/main/java/com/starrocks/sql/optimizer/operator/ColumnFilterConverter.java

```diff
@@ -80,12 +97,92 @@ private static boolean checkColumnRefCanPartition(ScalarOperator right) {
                 return type == columnType || (type != Type.LARGEINT && columnType != Type.LARGEINT);
             }
 
+            if (type.isDateType() && columnType.isDateType()) {
```

Comment:
if one type is date, another type is datetime, I think there will be some issues.